### PR TITLE
[MB-2246] Adds the ability to dismiss the default message center.

### PR DESCRIPTION
**Pull request by gatlinhebert**
Adds the ability to dismiss showing message center and message center messages. Dismissing message center is useful for apps that use the displayMessageCenter, while individual message dismissal is used for when an app displays their own message displaying.

On Android, its hard to access the activity stack in a clean non terrible way. Instead we will send an intent to the activities to close themselves. If the activity is currently running it will  receive the new intent through `onNewIntent`, if its not running it will just close itself immediately before its displayed to the user.

### CODE REVIEW

File: android/src/main/java/com/urbanairship/reactnative/CustomMessageCenterActivity.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            finish();
+            return;
+        }
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+
+        if (intent != null && "CLOSE".equals(intent.getAction())) {
```
Comment (crow) [inline android/src/main/java/com/urbanairship/reactnative/CustomMessageCenterActivity.java]: Probably can pull "CLOSE" out as a constant like:
    private static final String CLOSE_MESSAGE_CENTER = "CLOSE";

**Comment by crow** [comment]
🍔 manually tested on Android and iOS - looks good!

File: sample/AirshipSample/app.js [hunk 2]
```diff
@@ -260,42 +270,50 @@ export default class AirshipSample extends Component {
...
-        <EnableLocationCell
-          locationEnabled={this.state.locationEnabled}
-          handleLocationEnabled={this.handleLocationEnabled}
-        />
-        <Text style={styles.instructions}>
-          Press Cmd+R to reload,{'\n'}
-          Cmd+D or shake for dev menu
-        </Text>
-      </View>
+      <View style={styles.backgroundContainer}>
```
Comment (crow) [inline sample/AirshipSample/app.js]: This is me adding the view hierarchy into a scroll view and adding the message button functionality

**Comment by rlepinski** [comment]
👍

**Pull request merged**